packages/grid/x-data-grid/src/utils/createSelector.test.ts

```diff
@@ -9,6 +9,7 @@ describe('createSelector', () => {
     it('should warn if the instance ID is missing', () => {
       const selector = createSelectorMemoized([], () => []);
       const state = {} as GridStateCommunity;
+      // @ts-expect-error Need to test the warning
       expect(() => selector(state)).toWarnDev(
```

Comment:
Shouldn't this warning now be converted to an error?
Something like:
`'MUI: A selector was called without passing the instance ID, please pass instance ID as the second argument or pass apiRef instead.'`

Reply:
I think warning is fine, the goal of this PR was to require instanceId on a type level